ui/raidboss/data/03-hw/trial/sophia-ex.js

```diff
@@ -406,27 +437,33 @@
       netRegexCn: NetRegexes.startsUsing({ id: '19A9', source: '索菲娅' }),
       netRegexKo: NetRegexes.startsUsing({ id: '19A9', source: '소피아' }),
       condition: function(data) {
-        // No platform tilts if clones are up, and no tilts if it's the first one.
-        return !data.clonesActive && data.scaleSophias && data.sadTethers;
+        return data.sadTethers && data.scaleSophias;
       },
+      delaySeconds: .3,
       durationSeconds: 10,
       suppressSeconds: 5,
       alertText: function(data, matches) {
-        const safeDir = data.scaleSophias.indexOf(matches.sourceId) < 4 ? 'E' : 'W';
+        let safeDir = findSafeDir(data);
+        if (safeDir == 0)
+          safeDir = data.scaleSophias.indexOf(matches.sourceId) < 4 ? '2' : '-2';
         return {
-          'W': { en: 'Go West (Hard Tilt)', cn: '去西边（大倾斜）' },
-          'E': { en: 'Go East (Hard Tilt)', cn: '去东边（大倾斜）' },
-        }[safeDir];
+          '2': { en: 'Go East (Hard Tilt)', cn: '去东边（大倾斜）' },
```

Comment:
Can this be wrapped in a helper function too, rather than duplicating all this text?

Alternatively as a simplification proposal, do you think it'd be better to just put everything in the "cast" function even if it delayed callouts a little in some cases?

Reply:
We could wrap it easily, yes.

I feel like in all but three cases we *do* have the information to make the call solely from tether data. Unless it becomes really cumbersome, I think it might be better to keep them separated if we can?

Reply:
That's fair! I was just asking about the simplicity vs slightly earlier callout trade-off.  I think it's ok to call out earlier, I'd just prefer less repeated code and translations if there's some way to make that happen.